packages/amplify-graphql-auth-transformer/src/resolvers/query.ts

```diff
@@ -234,11 +236,28 @@ const generateAuthOnModelQueryExpression = (
             methodCall(ref('util.isNull'), ref('ctx.stash.authFilter')),
             not(ref('primaryFieldMap.isEmpty()')),
           ]),
-          compoundExpression([
+          sortKeyFields.length > 0
```

Comment:
Do we need this ternary operator here? The `IF` block `methodCall(ref('sortKeyFields.contains'), ref('entry.key')),` should take care of this check if it is initialized with an empty array.